public/components/Notifications/NotificationForm.tsx

```diff
@@ -168,18 +180,26 @@ export const NotificationForm: React.FC<NotificationFormProps> = ({
                   />
                 </EuiCompressedFormRow>
               </EuiFlexItem>
-              {prepareMessage && (
-                <EuiFlexItem>
-                  <EuiCompressedFormRow>
-                    <EuiSmallButton
-                      fullWidth={false}
-                      onClick={() => prepareMessage(true /* updateMessage */)}
-                    >
-                      Generate message
-                    </EuiSmallButton>
-                  </EuiCompressedFormRow>
-                </EuiFlexItem>
-              )}
+
+              <EuiFlexItem>
+                <EuiCompressedCheckbox
+                  id={`checked`}
+                  label={'Preview message'}
+                  checked={displayPreview}
+                  onChange={(e) => onDisplayPreviewChange(e)}
+                />
+              </EuiFlexItem>
+              {displayPreview ? (
+                <EuiCompressedFormRow label="Message preview" style={{ maxWidth: '100%' }}>
+                  <EuiCompressedTextArea
+                    placeholder="Preview of mustache template"
+                    fullWidth
+                    value={preview}
+                    readOnly
+                    className="read-only-text-area"
```

Comment:
Is this class used somewhere for styling? If not we can remove it

Reply:
Done